### ISSUE TITLE
Serialize string property as json object

### ISSUE THREAD

**Issue by ArtemMikhaylov** [Kotlin/kotlinx.serialization#2072]
**What is your use-case and why do you need this feature?**
I have a class like that
```kotlin
data class MyClass2(
    var id: Long,
    var name: String,
    var jsonData: String? = null
)
```
Property `jsonData` for saving json object as string, jsonData can be any json object with different contents

When sending class to server, I want to have MyClass2 mapped to json like this:

```json
{
    "id":7,
    "name":"Steve",
    "jsonData":{
         "someField1":true,
         "someField2":4,
         ...
    }
}
```
I can't figure out how to achieve this. I was able to do other way around: deserialize json object field to my class string property. But struggle with serialization.

**Describe the solution you'd like**

In GSON I used to achieve this by making JsonStringFieldAdapter and  annotation on property:
```kotlin
@JsonAdapter(JsonStringFieldAdapter::class)
var jsonData: String? = null
```
```java
public class JsonStringFieldAdapter implements JsonSerializer<String>, JsonDeserializer<String> {

    @Override
    public String deserialize(JsonElement json, Type typeOfT, JsonDeserializationContext context) throws JsonParseException {
        return json.toString();
    }

    @Override
    public JsonElement serialize(String src, Type typeOfSrc, JsonSerializationContext context) {
        return JsonParser.parseString(src).getAsJsonObject();
    }
}
```

I tried to make something similar with kotlinx.serialization
```kotlin
@Serializable(with = JsonAsStringSerializer::class)
var jsonData: String? = null
```
```kotlin
object JsonAsStringSerializer: JsonTransformingSerializer<String>(tSerializer = String.serializer()) {
    override fun transformDeserialize(element: JsonElement): JsonElement {
        return JsonPrimitive(value = element.toString())
    }

    override fun transformSerialize(element: JsonElement): JsonElement {
        return super.transformSerialize(element)
    }
}
```
Deserialization seems to be working, but I can't figure out how to do serialization

**Comment by ArtemMikhaylov** [comment]
Well, I could make it work, but in not very nice way. Maybe there's a nicer option? And if there's not, it would be cool to have something like in GSON (example above). 

That's how I made it work:
```kotlin
    override fun transformSerialize(element: JsonElement): JsonElement {
        return Json.decodeFromString<JsonObject>(
            element.toString()
                .replace("\\", "")
                .removePrefix("\"").removeSuffix("\"")
        )
    }
```
It looks like JsonElement representing String passed to `transformSerialize` , and it is already with slashes all around. It would be more flexible if during serialization we could map actual data type (in this case `String`) to desired JsonElement, and not dealing with already mapped JsonElement to transform it to our needs.

**Comment by aSemy** [comment]
This might be helped by a new feature that is yet to be released #2041, which would allow encoding a string directly, without surrounding it in quotes.
